### DISCUSSION TITLE
Consider removing the default `facetFilters` for DocSearch

### DISCUSSION THREAD

**Discussion by Lehoczky** [nuxt-modules/algolia#163] [Ideas]
I'm not an algolia pro, so I can be easily wrong here. The situation that led me to create this Discussion, is that I recently created a new DocSearch project, which automatically got indexed by the algolia crawler. When I installed this module, whatever I typed in the search, I never got any result, however when I tried out the basic example from the original docs, everything worked:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3"  />
  </head>
  <body>
    <div id="search"></div>

    <script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3"></script>
    <script type="text/javascript">
      docsearch({ appId: "...",  apiKey: "..", indexName: "...", container: "#search })
    </script>
  </body>
</html>
```

When I looked into the requests that my nuxt application sent, and what the basic example sent, the difference was that the nuxt app added an extra `facetFilters: ["language:en"]"` to the request headers. After some digging I found that my automatically indexed site had the `lang` attribute instead of `language`:

![image](https://github.com/nuxt-modules/algolia/assets/31937175/16dc081a-7fc4-44f6-a8fb-740df8ff490c)

However just changing the `langAttribute` in the options was not enough:

```ts
docSearch: {
   langAttribute: "lang",
   indexName: "...",
},
```

This still didn't solve the issue, because the language should have been `en-US` instead of the default `en`. It took me a lot of time to figure out that these are the options I need:

```ts
docSearch: {
   searchParameters: {
     facetFilters: ["lang:en-US"],
   },
   indexName: "...",
},
```

Finding these things out took a lot of time, and it feels like it was a lot harder to start a project with the nuxt module instead of just using the basic example I shared above. I think it would be better to align with the native behaviour and not send any extra `facetFilters`, or make this explicitly opt in. 
I also had to look into the module's source code to know how to solve my issue, which is not really the best first experience for developers, I feel like the docs was missing the information I looked for while debugging this.

**Comment by Lehoczky** [comment]
Might be related to this issue: https://github.com/nuxt-modules/algolia/issues/158

**Comment by Baroshem** [comment]
Hey

Thanks for this detailed research and explanation. I can completely understand that it was frustrating for you and I fully agree that there should be more information about it to avoid issues such as this one in the future.

However, I am not into removing the current behavior as this DocSearch is now being used in many web applications (including Nuxt documentation) so I would like to propose to maybe write a better documentation about what options are send by default and how user should operate to overwrite them.

Would that work for you? :)

**Comment by Lehoczky** [comment]
Thank you for your answer, Jakub!

The current behavour could be changed for the `2.0.0` release, where breaking changes will be allowed. Extending the documentation until then can be beneficial, but I feel like it would be only a "band aid solution".

I say this, because right now the module needs more setup than the official example. Ideally this module should also work out of the box, that would be the best developer experience.

Other than that, the default language filter, [which is English](https://github.com/nuxt-modules/algolia/blob/c4f576266a0d6ae7477228e0ec056d21aac1af32/src/runtime/components/AlgoliaDocSearch.vue#L73) is disadvantageous for anyone whose content is not in English (and you can see even in my case where the content is English but got indexed as `en-US` instead of `en`). These people would never see any search results without explicitly changing the search parameters and could feel that the defaults do not benefit them, instead just make their configuration harder.

And last but not least, aligning with algolia's default behavour would cause less thinking overhead when using this module. User's won't have to take a mental note and remember that the original client does not use `facetFilters` headers by default, but this module does.

I can totally be in the minority by thinking this, and I'm happy to hear what other users think about this implementation^^

**Comment by Baroshem** [comment]
Heyo,

Thanks for this detailed explanation. I have created an issue for adding the docs as a temporary solution https://github.com/nuxt-modules/algolia/issues/170 and a task for removing default facetFilters https://github.com/nuxt-modules/algolia/issues/169

The first one I will add in the upcoming days, while the second I will add to the scope for 2.0.0 release.

If you experience these kind of issues, other users may experience it as well so I am up for doing something with it. Let's also wait for some other comments :)
